Fix Network Error in Dev by Adding missing selfie analytics event

charleyf opened the pull request:
<!-- Uncomment and update the sections you need for your PR! -->

<!--
## 🎫 Ticket

Link to the relevant ticket:
[LG-XXXXX](https://cm-jira.usa.gov/browse/LG-XXXXX)
-->

## 🛠 Summary of changes

I'm looking into some black screen on open problems that selfie is happening. The first thing that caught my eye was a logger error. It's unrelated to the black screen, but this PR fixes that error.

## 📜 Testing Plan

Provide a checklist of steps to confirm the changes.

- [ ] On `main` get to the front/back/selfie upload page.
- [ ] Open Safari and use the "Develop" menu to view the console for the page you have open on your iPhone.
- [ ] Open to the "network" tab in safari so you can see the requests your iPHone is making.
- [ ] Click the `selfie` box.
- [ ] Note that there's a network request from `logger` that fails with a 400.
- [ ] Switch to this branch: `charley/fix-selfie-log`
- [ ] Repeat the steps above, there will now be no failed network requests from `logger`.

<!--
## 👀 Screenshots

If relevant, include a screenshot or screen capture of the changes.

<details>
<summary>Before:</summary>

</details>

<details>
<summary>After:</summary>

</details>
-->

aduth commented on app/services/analytics_events.rb:
```diff
@@ -2952,6 +2952,38 @@ def idv_selfie_image_added(
   # rubocop:enable Naming/VariableName,Naming/MethodParameterName
 
+  # rubocop:disable Naming/VariableName,Naming/MethodParameterName,
+  # rubocop:disable IdentityIdp/AnalyticsEventNameLinter
```
Let's name it per the rule and not disable this.

night-jellyfish commented on app/controllers/frontend_log_controller.rb:
```diff
@@ -33,6 +33,8 @@ class FrontendLogController < ApplicationController
     'IdV: personal key acknowledgment toggled' => :idv_personal_key_acknowledgment_toggled,
     'IdV: prepare submitted' => :idv_in_person_prepare_submitted,
     'IdV: prepare visited' => :idv_in_person_prepare_visited,
+    'IdV: selfie image added' => :idv_selfie_image_added,
```
This one is added below on [this line](https://github.com/18F/identity-idp/pull/10244/files#diff-15d821e4605e7b3b601472200fd81280ef62b62514314079b8d44ac6561716edR53).

Were you seeing it show with spaces? I wonder if I missed something in https://github.com/18F/identity-idp/pull/10208.

night-jellyfish replied:
I tested locally with both SDK and upload on `main` and didn't see any warnings about missing logs. Maybe that's not the error you're running into though.

charleyf replied:
Oh, interesting. I think I misunderstood the role of those two hashes. You're right that I don't actually care about `idv_selfie_image_added` for this PR. I just thought it was missing and added it.

I'm going to rework this PR shortly.

charleyf replied:
The error that I'm trying to fix is because we're missing `idv_selfie_image_clicked`

night-jellyfish commented on app/services/analytics_events.rb:
```diff
@@ -2952,6 +2952,38 @@ def idv_selfie_image_added(
+
   # Tracks when the user visits one of the the session error pages.
   # @param [String] type
   # @param [Integer,nil] submit_attempts_remaining (previously called "attempts_remaining")
```
just fyi, looks like CI is failing due to a missing changelog

night-jellyfish commented on app/controllers/frontend_log_controller.rb:
```diff
@@ -33,6 +33,8 @@ class FrontendLogController < ApplicationController
     'IdV: prepare submitted' => :idv_in_person_prepare_submitted,
     'IdV: prepare visited' => :idv_in_person_prepare_visited,
+    'IdV: selfie image added' => :idv_selfie_image_added,
+    'IdV: selfie image clicked' => :idv_selfie_image_clicked,
```
With this one you're inadvertently starting [LG-12178](https://cm-jira.usa.gov/browse/LG-12178) 😁

charleyf replied:
Mmmm, good point. But I want this now - it's causing confusing errors in the console 😆

night-jellyfish replied:
Yeah I think if it's causing issues in your current work - it makes sense to fix now! The ticket has some info on some properties we want on that event as well if it's easy. Otherwise if it's hard, we can save adding those properties for when the ticket is prioritized.

It's also a little bit different order than they're laid out, so we might have to update https://cm-jira.usa.gov/browse/LG-11982 to include this event once this is merged.

night-jellyfish commented on app/services/analytics_events.rb:
```diff
@@ -2952,6 +2952,37 @@ def idv_selfie_image_added(
+    source:,
+    use_alternate_sdk:,
+    liveness_checking_required: nil,
+    **_extra
```
Is `captureAttempts` easy to add in?

aduth replied:
> Is `captureAttempts` easy to add in?

👍 I'd hope we could add all of the properties here we're expecting to be logged.

aduth commented on app/services/analytics_events.rb:
```diff
@@ -2952,6 +2952,37 @@ def idv_selfie_image_added(
   end
   # rubocop:enable Naming/VariableName,Naming/MethodParameterName
 
+  # rubocop:disable Naming/VariableName,Naming/MethodParameterName,
```
For new analytics events, I'd wonder if it's worth trying to address some of what's being identified in these linters, i.e. naming the event properties with snake case. I suspect it might be made difficult by multiple events sharing the same common payload, but it seems like we will want to address that at some point anyways.

charleyf commented:
I'm going to leave a bit of time for comments, but my plan is to merge this shortly (~noon/1pm) as is and revisit in [LG-12178](https://cm-jira.usa.gov/browse/LG-12178).

I would prefer to add the correct properties to this event, remove the `rubocop:disable` and generally put more time into this, but I don't think that's where I should be putting my time right now.

Reasoning:
- I put this PR up because I want to remove a console error I'm seeing in dev that's making the black screen issue ([recent slack)](https://gsa-tts.slack.com/archives/C056RD1NEHW/p1710449522394939) more confusing.
- This event the way I implemented it matches the existing events `idv_front_image_clicked` and `idv_back_image_clicked`.
- The ticket here: https://cm-jira.usa.gov/browse/LG-12178
    - Contains the various improvements we want to make. I suspect we're going to need to make that improvement to all three of the `clicked` events: `idv_front_image_clicked`, `idv_back_image_clicked`, and idv_selfie_image_clicked`.
    - I've gone and updated that ticket with a link to this PR and also some notes about what we've discussed here.
